File: 26_remove_duplicates_from_sorted_array.py
'''
Given an integer array nums sorted in non-decreasing order, remove the duplicates in-place such that each unique element appears only once.
The relative order of the elements should be kept the same.
Then return the number of unique elements in nums.
'''
class Solution:
    # Time: O(n) / Space: O(1)
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0

        i = 0
        for j in range(1, len(nums)):
            if nums[j] != nums[i]:
                i += 1
                nums[i] = nums[j]
        return i + 1

File: typescript/26_remove_duplicates_from_sorted_array.ts
// Time: O(n) / Space: O(1)
function removeDuplicates(nums: number[]): number {
  if (!nums) return 0;

  let i = 0;
  for (let j = 1; j < nums.length; j++) {
    if (nums[j] != nums[i]) {
      i++;
      nums[i] = nums[j];
    }
  }
  return i + 1;
}
